packages/frontend/src/components/staking/StakingContainer.js

```diff
@@ -214,8 +214,8 @@ export function StakingContainer({ history, match }) {
         
         validators
             .filter((validator) => validator.version === FARMING_VALIDATOR_VERSION)
-            .forEach((validator) => dispatch(getValidatorFarmData(validator, accountId)));
-    }, [accountId, validators]);
+            .forEach((validator) => dispatch(getValidatorFarmData(validator, currentAccount.accountId)));
+    }, [currentAccount, validators]);
```

Comment:
```suggestion
    }, [currentAccount.accountId, validators]);
```
It's good to specify individual values rather than passing whole objects so we avoid running this effect when other values within the object we don't care about change.